sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIModelFactory.cs

```diff
@@ -7,29 +7,192 @@
+        /// </summary>
+        /// <param name="created">The timestamp to use for image generation origination</param>
+        /// <param name="data">The collection of ImageLocations to associate with the result</param>
+        /// <returns>A new instance of ImageGenerations</returns>
+        public static ImageGenerations ImageGenerations(
+            DateTimeOffset created,
+            IEnumerable<ImageLocation> data)
+        {
+            return new ImageGenerations(created, data);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of ContentFilterResult for tests and mocking.
+        /// </summary>
+        /// <param name="severity"> The severity level associated with this filter result </param>
+        /// <param name="isFiltered"> Whether or not the severity level for this filter result should block content </param>
+        /// <returns> A new instance of ContentFilterResult </returns>
+        public static ContentFilterResult ContentFilterResult(
+            ContentFilterSeverity severity = default,
+            bool isFiltered = default)
+        {
+            return new ContentFilterResult(severity, isFiltered);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of ContentFilterResults for tests and mocking.
+        /// </summary>
+        /// <param name="sexualFilterResult"> The content filter result for the 'sexual' category </param>
+        /// <param name="violenceFilterResult"> The content filter result for the 'violence' category </param>
+        /// <param name="hateFilterResult"> The content filter result for the 'hate' category </param>
+        /// <param name="selfHarmFilterResult"> The content filter result for the 'selfHarm' category </param>
+        /// <returns> A new instance of ContentFilterResults </returns>
+        public static ContentFilterResults ContentFilterResults(
+            ContentFilterResult sexualFilterResult = null,
+            ContentFilterResult violenceFilterResult = null,
+            ContentFilterResult hateFilterResult = null,
+            ContentFilterResult selfHarmFilterResult = null)
+        {
+            return new ContentFilterResults(sexualFilterResult, violenceFilterResult, hateFilterResult, selfHarmFilterResult);
+        }
+        /// <summary>
+        /// Initializes a new instance of PromptFilterResult for tests and mocking.
+        /// </summary>
+        /// <param name="index"> The index of the prompt to associate this filter result with </param>
+        /// <param name="contentFilterResults"> The content filter results associated with the matching prompt index </param>
+        /// <returns> A new instance of PromptFilterResult </returns>
+        public static PromptFilterResult PromptFilterResult(
+            int index = default,
+            ContentFilterResults contentFilterResults = null)
+        {
+            return new PromptFilterResult(index, contentFilterResults);
+        }
+
```

Comment:
I believe these five (ImageLocation, ImageGenerations, ContentFilterResult, ContentFilterResults, and PromptFilterResult) can be removed, as I see they are present in the auto-generated one?

Reply:
All part of that naming difference issue -- will answer on the converged comment (glad we caught this!)